File: CRUDCapacitacion-backend/src/main/java/com/example/CRUDCapacitacionbackend/services/CustomerService.java
package com.example.CRUDCapacitacionbackend.services;

import com.example.CRUDCapacitacionbackend.models.Customer;

import java.util.List;
import java.util.Set;

public interface CustomerService {

     Customer getCustomer(String id);
     Customer updateCustomer(Customer customer);
     Customer saveCustomer(Customer customer);

     List<Customer>getCustomers();

     void deteleCustomer(String id);
}

File: CRUDCapacitacion-frontend/src/app/service/customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import baserUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http:HttpClient) { }

  public listarCustomers(){
    return this.http.get(`${baserUrl}/customers/`);
  }

  public saveCustomer(customer:any){
    return this.http.post(`${baserUrl}/customers/`,customer);
  }

  public updateCustomer(customer:any){
    return this.http.put(`${baserUrl}/customers/`,customer);
  }
  
  public getCustomer(customerId:any){
    return this.http.get(`${baserUrl}/customers/${customerId}`);
  }

  public deteleCustomer(customerId:any){
    return this.http.delete(`${baserUrl}/customers/${customerId}`);
  }
}
